CMakeLists.txt

```diff
@@ -58,8 +58,19 @@ enable_vcpkg_manifest_feature(TEST_VAR COMPLEX_ENABLE_OrientationAnalysis FEATUR
 # --------------------------------------------------------------------------------------------------
 file(TO_CMAKE_PATH "${CMAKE_COMMAND}" CMAKE_COMMAND_NORM)
 
+# --------------------------------------------------------------------------------------------------
+# This section sets the version numbers for this build.
+# COMPLEX will use the normal semantic versioning system where the version is laid
+# out as MAJOR.MINOR.PATCH
+# --------------------------------------------------------------------------------------------------
+set(COMPLEX_VERSION_MAJOR 0)
+set(COMPLEX_VERSION_MINOR 1)
+set(COMPLEX_VERSION_PATCH 0)
+
+set(COMPLEX_VERSION "${COMPLEX_VERSION_MAJOR}.${COMPLEX_VERSION_MINOR}.${COMPLEX_VERSION_PATCH}")
```

Comment:
You actually don't have to individually define these variables. When the `project()` call is given a version, it automatically defines the following:
- `CMAKE_PROJECT_VERSION`
- `CMAKE_PROJECT_VERSION_MAJOR`
- `CMAKE_PROJECT_VERSION_MINOR`
- `CMAKE_PROJECT_VERSION_PATCH`

And they can also be accessed by using the project name directly. In our case it would be `complex_VERSION_MAJOR`, etc.